cc/attestation/cwt.cc

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2024 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cc/attestation/cwt.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "cc/attestation/cose.h"
+#include "libcppbor/include/cppbor/cppbor.h"
+#include "libcppbor/include/cppbor/cppbor_parse.h"
+
+namespace oak::attestation {
+
+absl::StatusOr<Cwt> Cwt::Deserialize(const std::vector<uint8_t>& data) {
+  // Deserialize COSE Sign1.
+  absl::StatusOr<CoseSign1> cose_sign1 = CoseSign1::Deserialize(data);
+  if (!cose_sign1.ok()) {
+    return cose_sign1.status();
+  }
```

Comment:
Are the usual macros available externally?

ASSIGN_OR_RETURN(absl::StatusOr<CoseSign1> cose_sign1, CoseSign1::Deserialize(data));

Reply:
Macros are not available externally